pkg/sdn/plugin/ovscontroller.go

```diff
@@ -254,16 +279,66 @@ func (oc *ovsController) SetPodBandwidth(hostVeth string, ingressBPS, egressBPS
 	return nil
 }
 
-func (oc *ovsController) UpdatePod(hostVeth, podIP, podMAC string, vnid uint32) error {
-	ofport, err := oc.ensureOvsPort(hostVeth)
+func getPodDetailsByContainerID(flows []string, containerID string) (int, string, string, string, error) {
+	note, err := getPodNote(containerID)
+	if err != nil {
+		return 0, "", "", "", err
+	}
+
+	// Find the table=20 flow with the given note and extract the podIP, ofport, and MAC from them
+	for _, flow := range flows {
+		parsed, err := ovs.ParseFlow(ovs.ParseForAdd, flow)
+		if err != nil {
+			return 0, "", "", "", err
+		}
+
+		if parsed.Table != 20 {
+			continue
+		}
+		noteAction, ok := parsed.FindAction("note")
+		if !ok || !strings.HasPrefix(noteAction.Value, note) {
```

Comment:
I've always felt guilty about assuming the note will always come back in lowercase (in ovscontroller.go:AlreadySetUp()) and was sort of hoping you'd cleverly avoid doing that here...